### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by rashmi-kulkarni-ct402**

### CODE REVIEW

File: Project3_STL/headers/Triangle.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "Point3D.h"
+
+namespace Shapes3D
+{
+    class Triangle1
```
Comment (adeshlad2) [inline Project3_STL/headers/Triangle.h]: Class name should be Triangle, not Triangle1

File: Project3_STL/headers/Triangle.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "Point3D.h"
+
+namespace Shapes3D
+{
+    class Triangle1
+    {
+    public:
+        Triangle1();
+        Triangle1(Point3D p1, Point3D p2, Point3D p3);
```
Comment (adeshlad2) [inline Project3_STL/headers/Triangle.h]: IDK but does it work without specifying namespace in parameters? It should Shapes3D::Point3D p1 and so on.

File: Project3_STL/headers/Triangle.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+namespace Shapes3D
+{
+    class Triangle1
+    {
+    public:
+        Triangle1();
+        Triangle1(Point3D p1, Point3D p2, Point3D p3);
+        ~Triangle1();
+
+        Point3D vertex1() const;
```
Comment (adeshlad2) [inline Project3_STL/headers/Triangle.h:13]: Same, add namespace.

File: Project3_STL/headers/Triangle.h (hunk 1 above)
Comment (rashmi-kulkarni-ct402) [reply Project3_STL/headers/Triangle.h]: > Class name should be Triangle, not Triangle1

I had a compiler issue yesterday so was trying with Vinayak if renaming classes might sort things out. I will rename Triangle class.

File: Project3_STL/src/Point3D.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
+#include "../headers/Point3D.h"
+using namespace Shapes3D;
+
+Point3D::Point3D()
```
Comment (adeshlad2) [inline Project3_STL/src/Point3D.cpp]: it is better/recommended to add initializer list in default constructor.

File: Project3_STL/headers/Triangle.h (hunk 2 above)
Comment (rashmi-kulkarni-ct402) [reply Project3_STL/headers/Triangle.h]: > IDK but does it work without specifying namespace in parameters? It should Shapes3D::Point3D p1 and so on.

The Triangle class in mentioned within the namespace so it'll work fine.
namespace Shapes3D
{
class Triangle 
{
-----
}

}

File: Project3_STL/src/STLReader.cpp [hunk 5]
```diff
@@ -0,0 +1,75 @@
+#include "../headers/STLReader.h"
+#include "../headers/Triangle.h"
+#include "../headers/Point3D.h"
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <vector>
+using namespace std;
+
+namespace Shapes3D
```
Comment (adeshlad2) [inline Project3_STL/src/STLReader.cpp]: Check if it works without using namespace block as it is already used in .h.

File: Project3_STL/src/Triangle.cpp [hunk 6]
```diff
@@ -0,0 +1,29 @@
+#include "../headers/Triangle.h"
+using namespace std;
+namespace Shapes3D
```
Comment (adeshlad2) [inline Project3_STL/src/Triangle.cpp]: Same, check without namespace.

**Review by adeshlad2 — CHANGES_REQUESTED** [review]

File: Project3_STL/src/STLReader.cpp (hunk 5 above)
Comment (rashmi-kulkarni-ct402) [reply Project3_STL/src/STLReader.cpp]: I have updated cpp files as per your comment and it is working fine.

File: Project3_STL/src/Triangle.cpp (hunk 6 above)
Comment (rashmi-kulkarni-ct402) [reply Project3_STL/src/Triangle.cpp]: I have updated cpp files as per your comment and it is working fine.

File: Project3_STL/src/Point3D.cpp (hunk 4 above)
Comment (rashmi-kulkarni-ct402) [reply Project3_STL/src/Point3D.cpp]: added initializer list for default constructor

**Pull request merged**